tests/PHPStan/Analyser/data/constant.php

```diff
@@ -0,0 +1,40 @@
+<?php // lint >= 8.1
+
+namespace Constant;
+
+use function PHPStan\Testing\assertType;
+
+define('FOO', 'foo');
+const BAR = 'bar';
+
+class Baz
+{
+	const BAZ = 'baz';
+}
+
+enum Suit
+{
+    case Hearts;
+}
+
+function doFoo(string $constantName): void
+{
+	assertType('mixed', constant($constantName));
+}
+
+assertType("'foo'", FOO);
```

Comment:
Can the asserts be moved into a  function or method?

Reply:
sure, but not sure if they really hurt here :)